Assets/Scripts/GameCharacter/GameCharacterAnimationController.cs

```diff
@@ -1,70 +0,0 @@
-using DG.Tweening;
-using System;
-using System.Collections;
-using System.Collections.Generic;
-using TMPro;
-using UnityEngine;
-
-public class GameCharacterAnimationController : GameCharacterComponent
-{
-
-    [SerializeField] TextMeshPro statusDisplay;
-    [SerializeField] MeshRenderer gunBarrelMeshRendered;
-    [SerializeField] MeshRenderer characterBodyMeshRendered;
-    [SerializeField] GameObject gunSmokeParticalEffect;
-    [SerializeField] Color attackAnimationColor, takeDamageAnimationColor;
-
-    GameCharacterController characterController;
-    Color gunBarrelColor,characterMaterial;
-   
-    private void Start()
-    {
-        gunSmokeParticalEffect.SetActive(false);
-        gunBarrelColor = gunBarrelMeshRendered.material.color;
-        characterMaterial = characterBodyMeshRendered.material.color;
-    }
-
-    public override void LoadCharacterDataToComponents(GameCharacterController controller)
```

Comment:
With any type of injection (in the constructor for a pure class or in Init for MB, you should always expect interfaces).
You inject implementations of course but you don't expect them.
For building the implementations I normally use a factory but you can have even a static Instantiator

Reply:
I'm not replicating this one all over the codebase but it's essentially true each time you see a similar case